File: 0167_twoSum.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        hashmap = {}
        for i, number in enumerate(numbers):
            if number not in hashmap:
                hashmap[target-number] = i + 1
            else:
                return [hashmap[number], i + 1]

File: 0001-0500/0167_twoSum.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int x = 0;
        int y = numbers.length - 1;

        while (true) {
            if (numbers[x] + numbers[y] == target) {
                return new int[] {x+1, y+1};
            } else if (numbers[x] + numbers[y] > target) {
                y -= 1;
            } else {
                x += 1;
            }
        }
    }
}
